internal/ecgit/ecgit.go

```diff
@@ -0,0 +1,306 @@
+		return nil, err
+	}
+	w, err := repo.Worktree()
+	if err != nil {
+		log.Debug("Unable to get repo worktree")
+		return nil, err
+	}
+	err = w.Checkout(&git.CheckoutOptions{
+		Branch: plumbing.ReferenceName(refName.String()),
+	})
+	if err != nil {
+		log.Debugf("Error checking out branch: %s\n", refName.String())
+		return nil, err
+	}
+	return w, nil
+}
+
+// applyDiff applies the diff to the specified repository.
+func applyDiff(repoPath string, diffFilePath string) error {
+	err := exec.Command("git", "-C", repoPath, "apply", diffFilePath).Run()
+	if err != nil {
+		log.Debugf("Error applying diff file '%s'\n", diffFilePath)
+		return err
+	}
+	return nil
+}
+
+// commitChange commits the changes in the worktree to the specified repository.
+func commitChange(repo *git.Repository, refName string, message string) (*object.Commit, error) {
+	w, err := repo.Worktree()
+	if err != nil {
+		log.Debug("Error getting repo worktree")
+		return nil, err
+	}
+	_, err = w.Add(".")
+	if err != nil {
+		log.Debug("Error adding files")
+		return nil, err
+	}
+	commit, err := w.Commit(message, &git.CommitOptions{
+		Author: &object.Signature{
+			Name:  "EC Automation",
+			Email: "enterprise-contract-service@redhat.com",
+			When:  time.Now(),
+		},
+	})
+	if err != nil {
+		log.Debug("Error creating commit in worktree")
+		return nil, err
+	}
+	obj, err := repo.CommitObject(commit)
+	if err != nil {
+		log.Debug("Error creating commit object")
+		return nil, err
+	}
+	return obj, nil
+}
+
+// pushChange pushes the changes to the specified repository.
+func pushChange(path string) error {
+	if os.Getenv("GITHUB_TOKEN") == "" {
+		log.Debug("Unable to push changes.")
+		return errors.New("GITHUB_TOKEN not set")
+	}
+	r, err := git.PlainOpen(path)
+	if err != nil {
+		log.Debugf("Unable to open repository in '%s'\n", path)
+		return err
+	}
+	auth := &http.BasicAuth{
+		Username: "EC Automation",
+		Password: os.Getenv("GITHUB_TOKEN"),
+	}
+	err = r.Push(&git.PushOptions{
+		Auth:  auth,
+		Force: true,
+	})
+	if err != nil {
+		log.Debug("Unable to push changes.")
+		return err
+	}
+
+	return nil
+}
+
+// createPullRequest creates a new pull request on the specified repository.
+func createPullRequest(client *github.Client, targetRepoOwner string, ctx context.Context, targetRepoName string, targetBranch string, sourceBranch string, prSubject string, prBody string, modifiable bool) (htmlurl string, err error) {
+	newPR := &github.NewPullRequest{
+		Title:               github.String(prSubject),
+		Head:                github.String(targetBranch),
+		Base:                github.String(sourceBranch),
+		Body:                github.String(prBody),
+		MaintainerCanModify: github.Bool(true),
+	}
+	if !modifiable {
+		newPR.MaintainerCanModify = github.Bool(false)
+	}
+	pr, _, err := client.PullRequests.Create(ctx, targetRepoOwner, targetRepoName, newPR)
+	if err != nil {
+		log.Debug("Unable to create pull request")
+		return htmlurl, err
```

Comment:
This could be a "naked" return since named parameters are used. Maybe named parameters are not a good fit for this function?